x/subscription/spec/03_messages.md

```diff
@@ -0,0 +1,73 @@
+<!--
+order: 3
+-->
+
+# Messages
+
+## MsgCreatePlan
+
+Merchant create plans, the message signer is the plan owner.
+
+```protobuf
+message MsgCreatePlan {
+  string title;
+  string description;
+  Coin price,  // amount + denomination
+  int32 expiration_time,
+  bool tolerate_failure,
+  int32 interval
+  int32 interval_offset
+}
+```
+
+## MsgStopPlan
+
+Sent by plan owner, also removes all subscriptions.
+
+```protobuf
+message MsgStopPlan {
+  int32 plan_id,
+}
+```
+
+## MsgCreateSubscription
+
+The message signer subscribe to the plan, fails if it'll expire before next collection time.
+
+```protobuf
+message MsgCreateSubscription {
+  int32 plan_id,
+  int32 expiration_time,
```

Comment:
ok for subscribers to choose expiration time @christopher-crypto ?

Reply:
changed to use duration configured in plan.